File: 백준/Silver/2606. 바이러스/바이러스.java
package baekjoon;
import java.util.*;


public class Main {
    static int[][] graph;
    static int[] visited;
    public static void bfs(int start){
        Queue<Integer> queue = new LinkedList<>();
        queue.add(start);
        visited[start] = 1;
        int q;
        int result = 0;
        while(!queue.isEmpty()){
            q = queue.remove();

            for(int i = 1; i<graph.length; i++){
                if(graph[q][i] != 0 && visited[i]!=1){
                    queue.add(i);
                    visited[i] = 1;
                    result += 1;
                }
            }

        }
        System.out.println(result);
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();

        graph = new int[n+1][n+1];
        visited = new int[n+1];
        int a, b;
        for(int i = 0; i<m; i++){
            a = sc.nextInt();
            b = sc.nextInt();

            graph[a][b] = 1;
            graph[b][a] = 1;
        }
        bfs(1);

    }
}

File: 백준/Silver/2606. 바이러스/바이러스.py
from collections import deque

def bfs():
  queue = deque()
  queue.append(1)

  while queue:
    q = queue.popleft()
    for g in graph[q]:
      if visited[g] == False:
        visited[g] = True
        queue.append(g)

  return sum(visited)-1


n = int(input())
m = int(input())
result = 0
visited = [False]*(n+1)
graph=[[] for _ in range(n+1)]
for _ in range(m):
  a,b = map(int, input().split())
  graph[a].append(b)
  graph[b].append(a)

print(bfs())